client/src/app/components/user-owned-stock-profile/user-owned-stock-profile.component.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+import { Router } from '@angular/router';
+import { Stock } from 'src/app/interfaces/stock';
+import { User } from 'src/app/interfaces/user';
+import { UserService } from 'src/app/services/user/user.service';
+
+export interface StockProfile {
+  company: string;
+  amount: string;
+  gain_loss: number;
+  price: string;
+  change: number;
+}
+@Component({
+  selector: 'app-user-owned-stock-profile',
+  templateUrl: './user-owned-stock-profile.component.html',
+  styleUrls: ['./user-owned-stock-profile.component.scss'],
+})
+export class UserOwnedStockProfileComponent implements OnChanges {
+  @Input() currentUser: User;
+  @Input() location: string;
+  @Input() userOwnedStocks: Stock[];
+  @Output() changeEarnings = new EventEmitter<{
+    earnings: number;
+    totalGain: number;
+    positive: boolean;
+  }>();
+  displayedColumns: string[] = [
+    'company',
+    'amount',
+    'gain_loss',
+    'price',
+    'change',
+  ];
+  public dataSource = new MatTableDataSource<StockProfile>([]);
+
+  constructor(private router: Router, private userService: UserService) {}
+  ngOnChanges(): void {
+    this.location === 'home'
+      ? (this.displayedColumns = ['company', 'amount', 'gain_loss'])
+      : (this.displayedColumns = [
+          'company',
+          'amount',
+          'gain_loss',
+          'price',
+          'change',
+        ]);
+    let data: StockProfile[];
+    data = [];
+    this.userService
+      .getUserAlpacaPosition(this.currentUser.id)
+      .then((result) => {
+        let earnings = 0;
+        let totalGain = 0;
+        result.forEach((r) => {
+          let amount = 0;
+          let quantity = 0;
+          let gain = 0;
+          let price = 0;
+          let change = 0;
+          this.userOwnedStocks
+            .filter((x) => x.tickerSymbol === r.symbol)
```

Comment:
We shouldn't show any stock with amount 0. Also the negative / positive colors are not working properly (it is red in all cases).